### PR TITLE
Add inspect subcommand to image

### PR DESCRIPTION
**Pull request by eiffel-fl**
Hi.

This PR adds `image inspect`:

```bash
$ sudo ./ig image inspect trace_exec -o jsonpretty          francis/image-inspect % u=
{
  "Repository": "ghcr.io/inspektor-gadget/gadget/trace_exec",
  "Tag": "latest",
  "Digest": "sha256:e13e3859be5ed8cef676a720274480d2748f66fd98cf8d963af6c4c05121526f",
  "Created": "2024-07-24T21:07:26Z"
}%
$ sudo ./ig image inspect trace_quux -o jsonpretty          francis/image-inspect % u=
Error: inspecting image: getting gadget image descriptor: Found tag "ghcr.io/inspektor-gadget/gadget/trace_quux:latest" but couldn't get a descriptor for it: not found
```

TODO:
- [ ]: Add documentation once we agree on the CLI.

Best regards.

### CODE REVIEW

**Review by alban — COMMENTED** [review]
Thanks for working on this!

I think the CLI for inspect is good.

I would like to add a way to get more information. I don't know if it needs another command or if it could be in the `inspect` command:
- list of architectures supported by the gadget
- list of layers
- information from gadget.yaml:
  - list of ebpf params
  - list of data sources with the list of fields
- information from the ebpf module
  - list of sections (so I can see kprobe/sys_mount)
  - list of ebpf maps with their types (so I don't have to use readelf and objdump)
  - show bpf bytecode + source (when available in ELF sections) of requested ebpf programs
- information from the wasm module
  - what `gadgetAPIVersion` does it use?
  - which wasm upcalls does it make?

Maybe part of it could be done later with different commands:
```
ig image layer list trace_exec
ig image layer inspect trace_exec application/vnd.gadget.ebpf.program.v1+binary
ig image layer extract trace_exec application/vnd.gadget.btfgen.v1+binary
```

File: cmd/common/image/inspect.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/columns/formatter/textcolumns"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/oci"
+
+	"golang.org/x/term"
+)
+
+func NewInspectCmd() *cobra.Command {
+	var outputMode string
+
+	outputModes := []string{utils.OutputModeColumns, utils.OutputModeJSON}
```
Comment (alban) [inline cmd/common/image/inspect.go:39]: OutputModeJSONPretty is missing

**Review by alban — COMMENTED** [review]
It needs documentation and a change on OutputModeJSONPretty. LGTM after that.

File: docs/reference/images.md [hunk 2]
```diff
@@ -270,3 +274,34 @@ $ sudo ig image list
 REPOSITORY                     TAG                           DIGEST       CREATED
 trace_open                     latest                        19ea8377298f 30 minutes ago
 ```
+
+#### `inspect`
+
+Inspect the given gadget image.
```
Comment (alban) [inline docs/reference/images.md:280]:
```suggestion
Inspect the given local gadget image.
```

AFAIU, inspect does not automatically pull an image if it does not exist locally.

File: docs/reference/images.md [hunk 3]
```diff
@@ -270,3 +274,34 @@ $ sudo ig image list
...
+
+```bash
+$ sudo ig image inspect -h
+Inspect the given gadget image
```
Comment (alban) [inline docs/reference/images.md:284]:
```suggestion
Inspect the given local gadget image
```

File: cmd/common/image/inspect.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+)
+
+func NewInspectCmd() *cobra.Command {
+	var outputMode string
+
+	outputModes := []string{utils.OutputModeColumns, utils.OutputModeJSON, utils.OutputModeJSONPretty}
+
+	cmd := &cobra.Command{
+		Use:          "inspect",
+		Short:        "Inspect the given gadget image",
```
Comment (alban) [inline cmd/common/image/inspect.go:43]:
```suggestion
		Short:        "Inspect the given local gadget image",
```

File: docs/reference/images.md [hunk 5]
```diff
@@ -88,9 +88,13 @@ Usage:
 
 Available Commands:
   build       Build a gadget image
+  export      Export the SRC_IMAGE images to DST_FILE
+  import      Import images from SRC_FILE
+  inspect     Inspect the given gadget image
```
Comment (alban) [inline docs/reference/images.md:93]:
```suggestion
  inspect     Inspect the given local gadget image
```

**Review by alban — APPROVED** [review]
One suggestion for clarification. LGTM after that.

File: cmd/common/image/inspect.go [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+	"github.com/docker/go-units"
+	"github.com/spf13/cobra"
+
+	"github.com/inspektor-gadget/inspektor-gadget/cmd/common/utils"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/columns"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/columns/formatter/textcolumns"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/oci"
+
+	"golang.org/x/term"
```
Comment (mauriciovasquezbernal) [inline cmd/common/image/inspect.go:33]: Please move it above. Didn't the CI catch this?

**Comment by mauriciovasquezbernal** [comment]
I'm reviewing it, please wait before merging.

**Review by mauriciovasquezbernal — APPROVED** [review]
Some minor comments but also LGTM!

**Pull request merged**